### PR TITLE
Fix when _stable_1d_sort to work when n >= N

### PR DESCRIPTION
**Pull request by frankier**
## What does this PR do?

Currently if n >= N we will get a "x_pad used before it is defined" type exception.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by SkafteNicki — APPROVED** [review]
LGTM

File: pytorch_lightning/metrics/utils.py [hunk 1]
```diff
@@ -286,5 +286,8 @@ def _stable_1d_sort(x: torch, N: int = 2049):
     if N - n > 0:
         x_max = x.max()
         x_pad = torch.cat([x, (x_max + 1) * torch.ones(2049 - n, dtype=x.dtype, device=x.device)], 0)
-    x_sort = x_pad.sort()
-    return x_sort.values[:n], x_sort.indices[:n]
+        x_sort = x_pad.sort()
+        return x_sort.values[:n], x_sort.indices[:n]
+    else:
+        x = x.sort()
+        return x.values, x.indices
```
Comment (carmocca) [inline pytorch_lightning/metrics/utils.py:293]:
```suggestion
    x_sort = x_pad.sort()
    i = min(N, n)
    return x_sort.values[:i], x_sort.indices[:i]
```

Comment (carmocca) [inline pytorch_lightning/metrics/utils.py:293]:
```suggestion
        x = torch.cat([x, (x_max + 1) * torch.ones(2049 - n, dtype=x.dtype, device=x.device)], 0)
    x_sort = x.sort()
    i = min(N, n)
    return x_sort.values[:i], x_sort.indices[:i]
```

Comment (frankier) [reply pytorch_lightning/metrics/utils.py:293]: If you prefer, sure

**Review by Borda — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by ananyahjha93 — APPROVED** [review]
